src/tools/blast/components/BlastForm.tsx

```diff
@@ -67,31 +67,32 @@ const getAutoMatrixFor = (sequence: string): FormParameters['matrix'] => {
 };
 
 const FormSelect: FC<{
-  formValues: BlastFormValues;
-  type: BlastFields;
-  updateFormValues: (type: BlastFields, value: string) => void;
-}> = ({ formValues, type, updateFormValues }) => {
-  const formObject = formValues[type];
-  if (!formObject) {
+  formValue: BlastFormValue;
+  updateFormValue: React.Dispatch<React.SetStateAction<BlastFormValue>>;
+}> = ({ formValue, updateFormValue }) => {
+  if (!formValue) {
     return null;
   }
+  const label = BlastFields[formValue.fieldName as keyof typeof BlastFields];
   return (
     <section className="blast-form-section__item">
-      <label htmlFor={type}>
-        {type}
+      <label htmlFor={label}>
```

Comment:
The label here is wrapping the corresponding select element, so no need to had extra ids and for attributes to link them.

Reply:
Done